dj/construction/utils.py

```diff
@@ -0,0 +1,104 @@
+"""
+Utilities used around construction
+"""
+
+from string import ascii_letters, digits
+from typing import TYPE_CHECKING, List, Optional, Set
+
+from sqlalchemy.orm.exc import NoResultFound
+from sqlmodel import Session, select
+
+from dj.construction.exceptions import CompoundBuildException
+from dj.errors import DJError, ErrorCode
+from dj.models.node import Node, NodeType
+
+if TYPE_CHECKING:
+    from dj.sql.parsing.ast import Namespace
+
+
+def make_name(namespace: Optional["Namespace"], name="") -> str:
+    """utility taking a namespace and name to make a possible name of a DJ Node"""
+    ret = ""
+    if namespace:
+        ret += ".".join(name.name for name in namespace.names)
+    if name:
+        ret += ("." if ret else "") + name
+    return ret
+
+
+def get_dj_node(
```

Comment:
Wait, we don't have a version of `get_dj_node` yet anywhere? @betodealmeida anything comes to mind?

Reply:
@agorajek when I read your comment I was convinced as well that there must be a very similar function somewhere! I couldn't find one but then I realized, since we're using SQLModel/sqlalchemy, it's so easy to just grab a node.
```py
query = select(Node).filter(Node.name =="foo")
session.exec(query).one()
```
That's easy enough to do which I figure is why a dedicated function never existed. This `get_dj_node` function here though adds two things that we'd want DRY from what I can tell.
- If a node can't be found (specifically, during a build), instead of raising an error, a `DJError` is added to a `CompoundBuildException`. This is important during a build since at the end we want to convert these errors into "missing parent" references that eventually get attached to a `draft` node.
- It optionally lets you include a `NodeType` filter and if a node by that name is found of a different type, it raises a very clear error.

Maybe @CircArgs can describe how he's using the second functionality right now but I do think that would have some use in future. For example I can see a common mistake by a user when looking up metrics by name is to see a transform node name and try and plug that into a metrics search. It's nice that we can return `Node foo is of type TRANSFORM. Expected kind to be of METRIC.`

Reply:
Thanks for the explanation @samredai . All makes sense. I don't think we need a better name for this method.